docs/nachet-inference-documentation.md

```diff
@@ -65,66 +68,224 @@ sequenceDiagram
+|labelOccurence | 0 | Contains the number of label occurence|
+|totalBoxes | 0 | Boxes total number|
+|Box | 1 | Contains all the information of one seed in the image|
+|label | 1 | Contains the top label for the seed|
+|score | 1 | Contains the top score for the seed|
+|topN | 1 | Contains the top N scores for the seed|
+|overlapping | 1 | Contains a boolean to tell if the box overlap with another one|
+|overlappingIndices | 1 | Contains the index of the overlapping box|
+|topX | 2 | The top x value of the box around a seed|
+|topY | 2 | The top y value of the box around a seed|
+|bottomX | 2 | The bottom x value of the box around a seed|
+|bottomY| 2 | The bottom y value of the box around a seed|
+
+*for more look at [nachet-model-documentation](https://github.com/ai-cfia/nachet-backend/blob/51-implementing-2-models/docs/nachet-model-documentation.md#return-value-of-models)*
+
+**topN** contains the top 5 predictions of the models:
+
+```json
+"topN": [
+    {
+        "label": "seed_name",
+        "score": 0.75
+    }
+    {
+        "label": "seed_name",
+        "score": 0.18
+    }
+    {
+        "label": "seed_name",
+        "score": 0.05
+    }
+    {
+        "label": "seed_name",
+        "score": 0.019
+    }
+    {
+        "label": "seed_name",
+        "score": 0.001
+    }
+]
+```
+
+### Blob storage and Pipeline versioning
+
+To keep track of the various pipeline iterations and versions, JSON files are
+stored in the blob storage. Users can add the JSON to the blob storage
+using the `pipelines_version_insertion.py` script. This allows for easy
+management of model and pipeline history.
+
+To use the script, 3 environment variables are necessary:
+
+* NACHET_BLOB_PIPELINE_NAME
+  * Containing the blob name where the pipelines are stored
+* NACHET_BLOB_PIPELINE_VERSION
+  * Containing the version the user wants to select
+* NACHET_BLOB_PIPELINE_DECRYPTION_KEY
+  * The key to decrypt sensible data such as the API key and the endpoint of a model.
+
+#### In the code
+
+In the backend, the pipelines are retrieved using the `get_pipelines` function.
+This function retrieves the data from the blob storage and stores the pipeline in
+the `CACHE["endpoint"]` variable. This the variable that feeds the frontend the `models`
+information and metadata.
+
+```python
+async def get_pipeline(mock:bool = False):
+    """
+    Retrieves the pipelines from the Azure storage API.
+
+    Parameters:
+    - mock (bool): If True, retrieves the pipelines from a mock JSON file. If False, retrieves the pipelines from the Azure storage API.
+
+    Returns:
+    - list: A list of dictionaries representing the pipelines.
+    """
+    if mock:
+        with open("mock_pipeline_json.json", "r+") as f:
+            result_json = json.load(f)
+    else:
+        result_json = await azure_storage_api.get_pipeline_info(connection_string, PIPELINE_BLOB_NAME, PIPELINE_VERSION)
+        cipher_suite = Fernet(FERNET_KEY)
```

Comment:
pass the cipher_suite and the result_json as the caller. don't rely on global environment variables being retrieved within your lower level functions.

"mock" should not be referenced within functions, that's a test facility

Reply:
I retrieve the mock here on PR55:

https://github.com/ai-cfia/nachet-backend/blob/51-implementing-2-models/docs/nachet-inference-documentation.md#in-the-code

Once PR55 is merge, I will correct it here with lower function receiving the environment variables

Reply:
Done

![image](https://github.com/ai-cfia/nachet-backend/assets/96267006/61000788-d709-4ddc-8270-05d6a097dc1a)

![image](https://github.com/ai-cfia/nachet-backend/assets/96267006/05c59c89-b077-4c11-8974-1f7980e56de6)